File: backend/src/main/java/com/vinicius/backend/domain/cao/dto/CaoResponse.java
package com.vinicius.backend.domain.cao.dto;

import com.vinicius.backend.domain.cao.enums.Genero;
import com.vinicius.backend.domain.cao.enums.StatusCao;
import com.vinicius.backend.domain.cao.enums.Tamanho;
import com.vinicius.backend.domain.cao.enums.TipoPelo;
import com.vinicius.backend.domain.imagem.dto.ImagemResponse;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public record CaoResponse(
        UUID id,
        String nome,
        TipoPelo tipoPelo,
        Tamanho tamanho,
        Genero genero,
        StatusCao status,
        LocalDate dataNascimento,
        String cor,
        String pedigree,
        String descricao,
        Boolean destaque,
        CaoPaisResponse pai,
        CaoPaisResponse mae,
        List<ImagemResponse> imagens,
        LocalDateTime criadoEm
) {}

File: frontend/src/services/caoService/caoResponse.ts
import type { ImagemResponse } from "../imageService/ImagemResponse";

export interface CaoResponse {
  id: string;
  nome: string;
  tipoPelo: string;
  tamanho: string;
  genero: string;
  status: string;
  dataNascimento: string;
  cor: string;
  pedigree: string;
  descricao: string;
  destaque: boolean;
  pai: { id: string; nome: string } | null;
  mae: { id: string; nome: string } | null;
  imagens: ImagemResponse[];
  criadoEm: string;
}
